vision/Object Detection/standard_obj_text.py

```diff
@@ -182,53 +227,229 @@ def _slice_rotate_img(self, img: np.ndarray, bounds: np.ndarray) -> np.ndarray:
 
         ## Rotate image ##
         rot_mat = cv2.getRotationMatrix2D(center_pt, angle, 1.0)
-        rotated_img = cv2.warpAffine(
+        self.rotated_img = cv2.warpAffine(
             cropped_img, rot_mat, cropped_img.shape[1::-1], flags=cv2.INTER_LINEAR
         )
 
-        return rotated_img
-
-    def _preprocess_img(self, img: np.ndarray) -> np.ndarray:
+    def _get_text_color(self, char_bounds: List[Tuple[int, int]]) -> Optional[str]:
         """
-        Preprocess image for text detection.
+        Detect the color of the text.
 
         Parameters
         ----------
-        img : np.ndarray
-            image to preprocess
+        char_bounds : List[Tuple[int, int]]
+            bounds of the text
 
         Returns
         -------
-        np.ndarray
-            the image after preprocessing
+        color : str
+            the color of the text
```

Comment:
I would reference in some way all the possible colors that can be output. If you want you can use the discrete output types as a set using `retstring : {"val1", "val2", "val3", "val4"}` instead of `retstring : str` or you can simply just reference your possible colors dictionary that you made.

Also, you need to mention here that color could be None (with `Optional[str]` or `str | None` and what a None type color logically means